### DISCUSSION TITLE
API /sp/targets/bid/recommendations how to use PAT_CATEGORY_REFINEMENT type ?

### DISCUSSION THREAD

**Discussion by LionSniper** [amzn/ads-advanced-tools-docs#240] [Q&A] [answered]
when request body is :
{
    "targetingExpressions": [
        {
            "type": "PAT_CATEGORY_REFINEMENT",
            "value": ""
        }
    ],
    "campaignId": "ABC123",
    "recommendationType": "BIDS_FOR_EXISTING_AD_GROUP",
    "adGroupId": "CDE123"
}

When the type in targetingExpressions is PAT_CATEGORY_REFINEMENT, what should be filled in the value? and what the difference between PAT_CATEGORY_REFINEMENT and PAT_CATEGORY ?

 I can not successfully test the case of PAT_CATEGORY_REFINEMENT, please give me an example, thanks！

**Comment by elizafarley** [comment, marked as answer]
Hi @LionSniper,

PAT_CATEGORY contains only a category ID, like:

```
"targetingExpression": {
   "type": "PAT_CATEGORY",
   "value": "123456"
}
```

PAT_CATEGORY_REFINEMENTS will return both the category, and possible refinements (like brand or price) in a single expression. Here's an example:

```
"targetingExpression": {
   "type": "PAT_CATEGORY_REFINEMENT",
   "value": "category=\"12456\" price=20-90"
}
```

Hope that helps!

**Reply by LionSniper** [reply]
Hi, @elizafarley
when value include age range key and genre key , how to use they?  please give me an example, thanks！

**Reply by LionSniper** [reply]
@elizafarley

**Reply by elizafarley** [reply]
Sure, in the recommendations endpoint, age range would look like this:

Request:
```
{
          "targetingExpression": {
            "type": "PAT_CATEGORY_REFINEMENT",
            "value": "category=\"12345\" age-range=\"54321\""
}
```

If you want to create a targeting expression based on this recommendation, you would call [POST /sp/targets](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/Targeting-clauses/operation/CreateSponsoredProductsTargetingClauses) using a request like this:

```
{
  "targetingClauses": [
    {
      "expression": [
        {
            "type":"ASIN_CATEGORY_SAME_AS",
            "value":"12345"
        },
        {
          "type": "ASIN_AGE_RANGE_SAME_AS",
          "value": "54321"
        }
      ],
      "campaignId": "11111111",
      "expressionType": "MANUAL",
      "state": "ENABLED",
      "bid": 1.50,
      "adGroupId": "2222222"
    }
  ]
}'
```

Genre would follow a similar pattern.

**Reply by rajatanand86** [reply]
Hello
I am looking at how to get bid recommendaitons for each keyword, can someone please help here?

**Reply by Alekxys29** [reply]
Hi @elizafarley , I'm currently having some difficulty figuring out how to format these targetDetails for the PAT_CATEGORY_REFINEMENT bid recommendation:

```
{
    "productCategoryId": "12345678",
    "productCategoryResolved": "Disposable Forks",
    "productBrand": "12345678",
    "productBrandResolved": "BRAND NAME HERE",
    "productPriceGreaterThan": "5",
    "productRatingLessThan": "5",
    "productRatingGreaterThan": "1",
    "productPrimeShippingEligible": true
}
```

Could you please provide an example of the correct formatting for this? Thanks a lot!

**Reply by elizafarley** [reply]
Are you looking for the format that will be returned by the recommendations endpoints, or the format you would use in the request to the /sp/targets endpoint?

**Reply by Alekxys29** [reply]
@elizafarley 

The format for the `value` that I will use in a request to `/sp/targets/bid/recommendations`, given the above values where type is `PAT_CATEGORY_REFINEMENT`.

**Reply by elizafarley** [reply]
Try this format:

```
{
"type": "PAT_CATEGORY_REFINEMENT",
"value": "category=\"12345\" brand=\"12345\" prime-shipping-eligible=\"true\" price=19.99-49.99 rating=4-5"
}
```
